File: src/domain/services/HistoryService.ts

import { OnAddedToDepartment } from "../events/OnAddedToDepartment";
import { History, NewHistory } from "../models/History";
import IHistoryRepository from "../repositoryInterfaces/IHistoryRepository";
import IHistoryService from "./IHistoryService";


class HistoryService implements IHistoryService {
  // Repository
  Repository: IHistoryRepository;

  constructor(repository: IHistoryRepository) {
    this.Repository = repository;
  }

  async createHistory(event: OnAddedToDepartment): Promise<History> {
    const newHistory = {
      ...event,
    } as NewHistory;

    const result = await this.Repository.createHistory(newHistory);
    return result;
  }
}

export default HistoryService;

File: src/wsample.domain/Services/HistoryService.cs
using wsample.domain.Infrastructure;
using wsample.domain.Models;

namespace wsample.domain.Services
{
    public class HistoryService : IHistoryService
    {
        private readonly IHistoryRepository _historyRepository;

        public HistoryService(IHistoryRepository historyRepository)
        {
            _historyRepository = historyRepository;
        }

        public async Task<int> CreateHistoryAsync(History newHistory)
        {
            // TODO map fields

            return await _historyRepository.CreateHistoryAsync(newHistory);
        }
    }
}
